File: ios/ui-framework-sample/UIFramework/Sources/UIFramework/BizLanguageSheet.swift
import SwiftUI

public struct BizLanguageSheetModifier: ViewModifier {
    @Binding var isPresented: Bool
    @Binding var selectedLanguage: String
    var onChange: ((String) -> Void)?
    
    let languages = [
        ("km", "ភាសាខ្មែរ", "🇰🇭"),
        ("en", "English", "🇬🇧"),
        ("ko", "한국어", "🇰🇷"),
        ("ja", "日本語", "🇯🇵"),
        ("zh", "漢語", "🇨🇳")
    ]
    
    public func body(content: Content) -> some View {
        content.bizBottomSheet(
            isPresented: $isPresented,
            content: {
                VStack(alignment: .leading, spacing: 0) {
                    Text("Please Select Your Language")
                        .font(.system(size: 16, weight: .bold))
                        .foregroundColor(Color(hex: "#1A2A5E"))
                        .padding(.bottom, 8)
                    
                    Text("Choose the language and region you want to use in smartBiz app")
                        .font(.system(size: 12))
                        .foregroundColor(Color(hex: "#555555"))
                        .lineSpacing(4)
                        .padding(.bottom, 24)
                    
                    VStack(spacing: 12) {
                        ForEach(languages, id: \.0) { lang in
                            LanguageOptionRow(
                                code: lang.0,
                                name: lang.1,
                                flag: lang.2,
                                isSelected: selectedLanguage == lang.0,
                                action: {
                                    selectedLanguage = lang.0
                                    onChange?(lang.0)
                                    DispatchQueue.main.asyncAfter(deadline: .now() + 0.2) {
                                        isPresented = false
                                    }
                                }
                            )
                        }
                    }
                }
                .padding(.horizontal, 24)
                .padding(.top, 8)
                .padding(.bottom, 32)
            }
        )
    }
}

private struct LanguageOptionRow: View {
    let code: String
    let name: String
    let flag: String
    let isSelected: Bool
    let action: () -> Void
    
    var body: some View {
        Button(action: action) {
            HStack {
                HStack(spacing: 12) {
                    Text(flag)
                        .font(.system(size: 20))
                    
                    Text(name)
                        .font(.system(size: 14, weight: .semibold))
                        .foregroundColor(Color(hex: "#111111"))
                }
                
                Spacer()
                
                ZStack {
                    Circle()
                        .stroke(isSelected ? Color(hex: "#1A2A5E") : Color(hex: "#EAEAEA"), lineWidth: 2)
                        .frame(width: 20, height: 20)
                    
                    if isSelected {
                        Circle()
                            .fill(Color(hex: "#1A2A5E"))
                            .frame(width: 10, height: 10)
                    }
                }
            }
            .padding(.horizontal, 16)
            .padding(.vertical, 12)
            .background(Color.white)
            .cornerRadius(12)
            .overlay(
                RoundedRectangle(cornerRadius: 12)
                    .stroke(isSelected ? Color(hex: "#1A2A5E") : Color(hex: "#EAEAEA"), lineWidth: 1.5)
            )
        }
    }
}

public extension View {
    func bizLanguageSheet(
        isPresented: Binding<Bool>,
        selectedLanguage: Binding<String>,
        onChange: ((String) -> Void)? = nil
    ) -> some View {
        self.modifier(BizLanguageSheetModifier(
            isPresented: isPresented,
            selectedLanguage: selectedLanguage,
            onChange: onChange
        ))
    }
}

File: android/ui-framework-sample/ui-framework/src/main/java/kh/mcnc/lib/components/BizLanguageSheet.kt
package kh.mcnc.lib.components

import androidx.compose.foundation.BorderStroke
import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.delay
import kotlinx.coroutines.launch

@Composable
fun BizLanguageSheet(
    isVisible: Boolean,
    selectedLang: String,
    onDismissRequest: () -> Unit,
    onChange: (String) -> Unit
) {
    data class LangOption(val code: String, val name: String, val flag: String)
    val languages = listOf(
        LangOption("km", "ភាសាខ្មែរ", "🇰🇭"),
        LangOption("en", "English", "🇬🇧"),
        LangOption("ko", "한국어", "🇰🇷"),
        LangOption("ja", "日本語", "🇯🇵"),
        LangOption("zh", "漢語", "🇨🇳")
    )

    BizBottomSheet(
        isVisible = isVisible,
        onDismissRequest = onDismissRequest
    ) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(horizontal = 24.dp)
                .padding(top = 8.dp, bottom = 32.dp)
        ) {
            Text(
                text = "Please Select Your Language",
                fontSize = 16.sp,
                fontWeight = FontWeight.Bold,
                color = Color(0xFF1A2A5E),
                modifier = Modifier.padding(bottom = 8.dp)
            )
            Text(
                text = "Choose the language and region you want to use in smartBiz app",
                fontSize = 12.sp,
                color = Color(0xFF555555),
                lineHeight = 16.sp,
                modifier = Modifier.padding(bottom = 24.dp)
            )

            Column(
                modifier = Modifier.fillMaxWidth(),
                verticalArrangement = Arrangement.spacedBy(12.dp)
            ) {
                languages.forEach { lang ->
                    val isActive = selectedLang == lang.code
                    Surface(
                        shape = RoundedCornerShape(12.dp),
                        color = Color.White,
                        border = BorderStroke(
                            width = 1.5.dp,
                            color = if (isActive) Color(0xFF1A2A5E) else Color(0xFFEAEAEA)
                        ),
                        modifier = Modifier
                            .fillMaxWidth()
                            .clickable {
                                onChange(lang.code)
                                CoroutineScope(Dispatchers.Main).launch {
                                    delay(200)
                                    onDismissRequest()
                                }
                            }
                    ) {
                        Row(
                            modifier = Modifier
                                .fillMaxWidth()
                                .padding(horizontal = 16.dp, vertical = 12.dp),
                            horizontalArrangement = Arrangement.SpaceBetween,
                            verticalAlignment = Alignment.CenterVertically
                        ) {
                            Row(
                                verticalAlignment = Alignment.CenterVertically,
                                horizontalArrangement = Arrangement.spacedBy(12.dp)
                            ) {
                                Text(
                                    text = lang.flag,
                                    fontSize = 20.sp
                                )
                                Text(
                                    text = lang.name,
                                    fontSize = 14.sp,
                                    color = Color(0xFF111111),
                                    fontWeight = FontWeight.SemiBold
                                )
                            }

                            Box(
                                modifier = Modifier
                                    .size(20.dp)
                                    .background(Color.Transparent, CircleShape),
                                contentAlignment = Alignment.Center
                            ) {
                                Surface(
                                    modifier = Modifier.fillMaxSize(),
                                    shape = CircleShape,
                                    color = Color.Transparent,
                                    border = BorderStroke(
                                        width = 2.dp,
                                        color = if (isActive) Color(0xFF1A2A5E) else Color(0xFFEAEAEA)
                                    )
                                ) {}
                                if (isActive) {
                                    Box(
                                        modifier = Modifier
                                            .size(10.dp)
                                            .background(Color(0xFF1A2A5E), CircleShape)
                                    )
                                }
                            }
                        }
                    }
                }
            }
        }
    }
}
